model/metricset.go

```diff
@@ -199,15 +199,16 @@ func (me *Metricset) Transform(ctx context.Context, cfg *transform.Config) []bea
 	fields["processor"] = metricsetProcessorEntry
 
 	if cfg.DataStreams {
+		dataset := AppMetricsDataset
 		// Metrics are stored in "metrics" data streams.
 		if isInternal {
 			// Metrics that include well-defined transaction/span fields
 			// (i.e. breakdown metrics, transaction and span metrics) will
 			// be stored separately from application and runtime metrics.
-			fields[datastreams.DatasetField] = InternalMetricsDataset
-		} else {
-			fields[datastreams.DatasetField] = AppMetricsDataset
+			dataset = InternalMetricsDataset
 		}
+		dataset += "." + datastreams.NormalizeServiceName(me.Metadata.Service.Name)
```

Comment:
nit: Could you also use `fmt.Sprintf` here as everywhere else; it would only create one new string, instead of two.